compiler/ghc/default.nix

```diff
@@ -33,14 +33,18 @@
 
 , # Whether to build dynamic libs for the standard library (on the target
   # platform). Static libs are always built.
-  enableShared ? !stdenv.targetPlatform.isWindows && !stdenv.targetPlatform.useiOSPrebuilt
+  enableShared ? stdenv.targetPlatform == stdenv.hostPlatform
+    # Allow enableShared for musl ghci support
+    || (stdenv.hostPlatform.isLinux && stdenv.targetPlatform.isMusl)
```

Comment:
Again the same `crossCompiler` logic. Maybe we should have a predicate `isCrossCompiler(stdenv)` instead?